### ISSUE TITLE
Prevent transgender harassment

### ISSUE THREAD

**Issue by estelle** [travisbrown/memory.lol#3]
Would be cool to add a flag that if the bio contains the transgender flag or pronouns, the script fails. While people will be able to look at the script and edit out this feature, it will help prevent non-technical trolls / lazy bullies from surfacing dead names.

**Comment by LuNeder** [comment]
Yeah I can definitely see how this tool could easily backfire and be used for harassing trans people 

Also, it’d be nice to add something hidden to the bio instead of the trans flag
Maybe just the LGBT flag
Some people out there might not want to be publicly trans

**Comment by LuNeder** [comment]
And honestly, I think this will probably be used MORE by transphobes to harass transgender people than to unmask transphobes. Big part of the transphobes already have their real name out anyway or just create fake accounts, while lots of transgender people I follow and am friends with just keep their account and change their usernames.

And this having already more than 100 stars, it might be too late to avoid this. While I do like to imagine that most of these stars have not downloaded this yet, they can always download a previous commit.

While having a protection would be better than nothing, I’d assume the best thing would be just privating this as a whole to avoid harassment. One techy transphobe would be enough to remove the protection and redistribute this among all other transphobes.

**Comment by hyperfekt** [comment]
Mind that this service relies upon a database that contains the data from the Wayback Machine (easy to access, and censorable by users upon request) and the Twitter Stream Grab (hard to access, unclear if it would be redacted). This repo alone is useless, so the question should be about public access to the database and the general problem of the Internet Archive maintaining a Twitter history.
In my opinion it might be advisable to keep access to the database only upon request, and for the community to look into whether the Stream Grab can be changed (Travis would still be able to steward a copy of the unredacted data, which could be useful as people with more nefarious intent would likely also want to have their data removed from the Stream Grab).

**Comment by travisbrown** [comment]
Thank you for raising this issue here. A few notes:

* Most of the information the service provides can easily be found through other means (Twitter search, the Wayback Machine). People who engage in harassment tend to know about these methods. The goal of the project is to make it easy for anyone to identify accounts that may be abusive, scammy, spreading disinformation, etc. It doesn't change what is possible (either for "good" use cases or abusive ones).
* The underlying service has been publicly available (and publicized) for a month, and has been in use privately by several groups of antifascist researchers and journalists since last November. I've been watching for misuse, and what I've seen has been very limited. If that changes we will restrict access to the service.

There is also a private exclusion list that removes ID-screen name pairs from the public results (although of course the excluded info is usually still also available through Twitter search and the Internet Archive's Twitter Stream Grab, which I have no control over). I will respond privately to all requests for exclusions.

**Comment by travisbrown** [comment]
@hyperfekt As far as I know, nothing has ever been redacted from the Twitter Stream Grab (although redactions regularly happen for the Wayback Machine), and the way it's packaged and distributed would make specific redactions practically impossible.

**Comment by travisbrown** [comment]
One other thing to note is that (as @hyperfekt points out above), the availability of the code in this repository isn't very useful on its own—it takes a lot of computing time to build the index from the Twitter Stream Grab and Wayback Machine, and the scraping code isn't available here. Access to the web service can be restricted (or turned off entirely), and without it the browser extension is useless.

**Comment by UwUnyaa** [comment]
Just to let you know, this whole project is so bad that I had to look up how to report repositories on GitHub. I refuse to call it a tool, because it's nothing but a weapon.

**Comment by travisbrown** [comment]
@UwUnyaa In what sense is an index of public screen names from public archives (which are in most cases still discoverable through Twitter's public interface) a "weapon"?

**Comment by travisbrown** [comment]
I've just restricted access to the web service, which means that the Chrome extension will no longer add previous screen names to Twitter pages for now. If you had access to the service before the public launch last month, feel free to contact me.

Please note that all of the data indexed by the service is **very** public. If you feel that information provided by the service is a threat to your safety, you should be aware that there are several other ways for people to find that info (in most cases extremely easily, via a single Twitter search), and that there are steps that you should be taking to identify (and possibly limit) your exposure. I'm happy to discuss those steps, or to put you into contact with someone else who can help.

I'll post an update here before opening the service to the public again, and in the meantime please see the [README](https://github.com/travisbrown/memory.lol#safety) for information on requesting exclusions.

**Comment by UwUnyaa** [comment]
@travisbrown Making this kind of OSINT easier is just a bad idea, especially with the current state of affairs.

**Comment by LuNeder** [comment]
> If it's already public and people are able to get around it by making a new account or something, I don't see the issue. The corporations already have access to all the data. It's already possible for people who are intent on doxxing/harrassing, making it easier for regular people vs institutions just seems clearly helpful.

Well, it being public doesn't mean one will know how to do it. This thing makes it terribly easy to do it, anyone would be able to use this fore harassing.

And creating a new account is not something most people want, specially if you already have lots of followers that you would end up losing or if you already have your account for a long time.

Still, now that this tool exists, the only option for lots of people may very well be creating a new account. The exclusion thing will definitely help, but now that it's public that doing this can be easy one techy transphobe can just create an unrestricted alternative.

Twitter themselves should probably change how they deal with accounts uuids. Probably making them private, being only something to deal with suspensions and bans and stuff, not something public that everyone can see.

**Comment by Howlsyawn** [comment]
Trustworthy journalists? And how does that work? Who decides who's trustworthy? If you want your data to be private you should understand that the internet is a giant melting pot of data and it can either be available to all or available to some.

**Comment by LuNeder** [comment]
> @LuNeder They already harass plenty, you can block. 

Yeah they already harass plenty, why not make it worse right?

> The question seems to be about deadnames primarily, 

Not really. People who want to stay anonymous for any other reason, such as abusive family, will be affected.

> which seems to me like something worth making a new account for when the data is public, or tolerating that some people will know. 

As I said, some people have had an account for a long time or have lots of followers. Creating a new account will make you lose old tweets and lots of followers as well. And the data was never easily publicly accessible until now. Even I, a person who would be completely able to access this data manually if I wanted, did not know that it was possible. This tool will make it a thing for everyone.



> Everyone of those followers a trans person had before switching names would know about their previous identity,

Usually they won't go harassing you or making this information public tho.


> The data exists. That's just what happens when you share your information with people and build an identity that you then want to change. 

Once again, until now you were able to change it. Username changes exist for a reason. If anyone can easily see your old username then there's no reason for them to even exist.

> Do we really want twitter to be in control of all our data and put it all behind closed doors? 

I don't get how a private random UUID would be controlling your data, and how making things you don't want to be public private would be a bad thing.

**Comment by Howlsyawn** [comment]
Whatever. Just delaying the inevitable seems like to me. Making it harder for people with less resources and easier for people with more resources.

**Comment by travisbrown** [comment]
> This is not a job for random internet users.

@LuNeder To take a different example: early in the escalation of the Russian invasion of Ukraine in February there were a number of "OSINT" accounts that quickly gained large audiences (100k+ followers) and then turned out either to be scams or to be used for disinformation. Several of them were former NFT or novelty accounts that were bought or repurposed because they had established followings (and were presumably less likely to get suspended because they had histories). This kind of thing is extremely common on Twitter, and if you have a list of former screen names on the profile page, it's often really easy to identify.

> Going on a json of the size of twitter itself is not something anyone will do

The Twitter Stream Grab isn't the only way to find the information indexed by this service (and it's not even that difficult to work with if you know particular date ranges to target). It's also possible to construct queries with the Twitter advanced search interface that are extremely effective at surfacing old screen names. This tool doesn't have anything to do with that method, which was well known before it was launched, and which will continue to be used by attackers.

**Comment by travisbrown** [comment]
> If people can't learn to stop oversharing and being attached to accounts held by giant unaccountable corporations then idk how giving them safety rails everywhere is helping them. They need to make new accounts and learn digital literacy.

@Howlsyawn I'm sorry but I don't think this tone is helpful here. I agree that it would be good for people to be more aware of threats to their security or privacy, but I don't think framing the issue in terms of blame or guilt is productive.

**Comment by Howlsyawn** [comment]
I didn't blame them. The problem is that the systems we have are not really under our control anyways. I said that helping them would not involve giving them more safety rails to delay the inevitable. It would entail giving them digital literacy and pressuring the companies that actually control the data to give us more control as users.

**Comment by valknight** [comment]
This tool has a purpose, but I think it's irresponsible to make it so easy to use, with guides on how to pull the data into spreadsheets with wildcard matching, when doxxing forums like Kiwifarms exist.

I feel the more appropriate option here would be to pull the API for now, and require application for researchers and journalists to use it, or even handing this project off to a non-profit org that can handle access to the tool. I think one person having control of this tool is unwise and unsafe.

**Comment by lizthegrey** [comment]
+1 to "show recent name changes only" e.g. those made in the past 2 months, that should address the scammy buy account with loads of followers and rename situation.

**Comment by travisbrown** [comment]
@lizthegrey That sounds reasonable. It wouldn't support many use cases (e.g. projects like finding deleted tweets by January 6 participants), but the full histories could be provided to authenticated (invited) users.

**Comment by commieosint** [comment]
FWIW, there is evidence that fash have a similar service that they've been using for several months now. Not sure how much its been shared within their circles or if a similar process has been used to make it

**Comment by NireBryce** [comment]
All of that said, if you have concerns about you personally being targeted directly, the people doing the targeting know all the moving pieces here.  

The best you can do is pressure Twitter to remove your info, or move accounts which, ime, is never a thing you can actually fully keep your following from, or your reputation.

and that's the part of this whole thing that sucks the most.

I think it's important for antifascist work, but I also think addition of names should require them being directly added to the dossier database instead of being scraped in full.  At least then you can remove everything a malicious account added if you find they're doing malicious searches, instead of them just being there for all to see without a login

**Comment by travisbrown** [comment]
> The browser extension as it is currently designed does not provide any additional benefit in the use cases mentioned above, while potentially adding to harm regarding deadnames.

I agree that the browser extension was a mistaken experiment in its current form, and I take responsibility for that mistake. If it's reinstated for general use, it will be in a more limited form that's designed to support a narrower set of use cases (probably only showing recent screen name changes, as discussed above).

**Comment by travisbrown** [comment]
I'm aiming to roll out the plan above this weekend. To summarize: screen name changes from the last 60 days will be visible to all users, with the full index only being available to trusted users (currently only supporting authentication through GitHub, but I might add Google before deploying it).

**Comment by travisbrown** [comment]
The service is [now publicly available](https://github.com/travisbrown/memory.lol#update-on-access-15-july-2022), but unauthorized users will only be shown screen names that were observed in the past week. I'll bump that up to 60 days on Monday unless someone raises serious concerns. 60 days should be enough for partial support of one of the main use cases for the service (identifying scams and disinformation).

The full index is also now available to a small trusted group who can sign in via GitHub. The service only uses GitHub for authentication, doesn't require any non-public or write access to the user's GitHub account, will never request any kind of password, and only the user's GitHub ID is stored on the servers.

The service does not currently log requests in a way that would allow me to link individual queries to specific authorized users, but I reserve the right to implement such logging in the future if there's any suggestion of abuse.

**Comment by estelle** [comment]
Thank you!

**Comment by HolobotHowlsyawn** [comment]
never mentioned blame or guilt
